pkg/certificate/controller_test.go

```diff
@@ -223,6 +226,31 @@ var _ = Describe("Certificates controller", func() {
 				isTLSEventuallyVerified().Should(Succeed(), "should eventually have a TLS secret")
 			})
 		})
+		Context("and TLS secret is deleted using cascade deletion", func() {
+			BeforeEach(func() {
+				// Emulate cascade deletion by breaking ownership and
+				// setting it to a non existing one
+				By("Break TLS secret ownership")
+				obtainedSecret := corev1.Secret{}
+				err := cli.Get(context.TODO(), types.NamespacedName{Namespace: expectedSecret.Namespace, Name: expectedSecret.Name}, &obtainedSecret)
+				Expect(err).To(Succeed(), "should succeed getting the secret")
+				obtainedSecret.OwnerReferences[0].Name = "bad-service"
+				err = cli.Update(context.TODO(), &obtainedSecret)
+				Expect(err).To(Succeed(), "should succeed updating the TLS secret")
+
+				By("Wait a little for Reconcile to settle after updating secret")
+				time.Sleep(3 * time.Second)
+
+				By("Delete the TLS secret")
```

Comment:
can you explain how does this mimic cascade delete of the secret? cascase is when the object is delete indirectly by deleting its owner, no?